### PR TITLE
Refactor CI, introduce examples

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

Create real examples to use pupernetes in SaaS CI.

### Motivation

The current CI configuration for pupernetes itself create noise around the testing of the project itself.

I seized this opportunity to improve the current CI system too.

### CODE REVIEW

**Comment by JulienBalestra** [comment]
I discovered some issues with a command line with pupernetes in `$PATH` see #72

**Comment by JulienBalestra** [comment]
@kindermoumoute @CharlyF this PR needs to be merge when we release `0.6.0`.

The `--kubeconfig-config` flag was introduced by #69 and that's not released yet.

File: examples/circleci.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      DEBIAN_FRONTEND: "noninteractive"
+
+jobs:
+  pupernetes:
+    <<: *machine_job_template
+    steps:
+      - checkout
+      - run:
+          name: download
+          command: sudo curl -Lf https://github.com/DataDog/pupernetes/releases/download/v0.6.0/pupernetes -o /usr/local/bin/pupernetes && sudo chmod +x /usr/local/bin/pupernetes
```
Comment (kindermoumoute) [inline examples/circleci.yaml]: v0.6.1

File: examples/travis.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    - HYPERKUBE_VERSION=1.8.14
+
+before_install:
+  - sudo apt-get update -qq
+
+install:
+  - sudo apt-get install -yqq systemd
+  - sudo curl -Lf https://storage.googleapis.com/kubernetes-release/release/v$HYPERKUBE_VERSION/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl
+  - sudo chmod +x /usr/local/bin/kubectl
+  - sudo curl -Lf https://github.com/DataDog/pupernetes/releases/download/v0.6.0/pupernetes -o /usr/local/bin/pupernetes
```
Comment (kindermoumoute) [inline examples/travis.yaml]: v0.6.1

**Pull request merged**